examples/gdpr-cookie-consent/app/root.tsx

```diff
@@ -0,0 +1,69 @@
+import {
+  Links,
+  LiveReload,
+  LoaderFunction,
+  Meta,
+  Outlet,
+  Scripts,
+  ScrollRestoration,
+  useLoaderData,
+  ActionFunction,
+  redirect,
+  Form
+} from "remix";
+import { gdprConsent } from "./cookies";
+
+export const loader: LoaderFunction = async ({request}) => {
+  const cookieHeader = request.headers.get("Cookie");
+  const cookie = (await gdprConsent.parse(cookieHeader)) || {};
+  return { track: cookie.gdprConsent };
+}
+
+export const action: ActionFunction = async ({request}) => {
+  const formData = await request.formData();
+  const cookieHeader = request.headers.get("Cookie");
+  const cookie = (await gdprConsent.parse(cookieHeader)) || {};
+
+  if (formData.get("accept-gdpr") === "true") {
+    cookie.gdprConsent = true;
+  }
+
+  return redirect("/", {
+    headers: {
+      "Set-Cookie": await gdprConsent.serialize(cookie)
+    }
+  });
+}
+
+export default function App() {
+  const {track} = useLoaderData();
+  return (
+    <html lang="en">
+      <head>
+        <meta charSet="utf-8" />
+        <meta name="viewport" content="width=device-width,initial-scale=1" />
+        <Meta />
+        <Links />
+      </head>
+      <body>
+        <Outlet />
+        {!track && <div style={{
+      backgroundColor: '#ccc',
+      padding: 10,
+      position: 'fixed',
+      bottom: 0,
+     }}>
+        <Form method="post" reloadDocument>
```

Comment:
Can we avoid the use of `reloadDocument`?

Reply:
Without it the code on public/dummy-analytics-script.js will not run immediately after the form submission.
Is there a better way of doing it?

Reply:
Yeah, looks like inserting a script tag with React doesn't work, so we have to use `document.createElement` in a `useEffect` which is fine. I've made that update myself (hope you don't mind) and I also changed it to use `useFetcher` so we don't get a navigation and add to the history stack.

Reply:
Thank you :)